app/components/DropdownWithModal/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC } from "react";
+import Image from "next/image";
+import { Trans } from "@lingui/macro";
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+
+import { Text } from "@klimadao/lib/components";
+
+import * as styles from "./styles";
+
+interface Item {
+  name: string;
+  icon: StaticImageData;
+  balance?: string;
+  disabled?: boolean;
+}
+
+interface Props {
+  modalTitle: string;
+  label: string;
+  currentItem: Item;
+  items: Item[];
+  isModalOpen: boolean;
```

Comment:
can the modal state be held inside this component, since this is where the modal is rendered? Or does something else need to control it outside of this dropdown?